src/coreclr/src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1242,33 +1644,35 @@ internal static bool IsAttributeDefined(RuntimeModule decoratedModule, int decor
                         // big-endian platforms.
 #if BIGENDIAN
                         const int CustomAttributeVersion = 0x0100;
+                        int version = Unsafe.ReadUnaligned<short>((void*)blobStart);
+                        if (version != CustomAttributeVersion)
+                        {
+                            throw new CustomAttributeFormatException();
+                        }
+                        cNamedArgs = BinaryPrimitives.ReverseEndianness(Unsafe.ReadUnaligned<short>((byte*)blobStart + 2));
```

Comment:
NumNamed field in the custom attribute blob is `unsigned int16`. (The code before this change did not have it correctly either.)